### PR TITLE
Add some more basic Http2 client tests

### PR DESCRIPTION
**Pull request by raboof**
Part of #3227 but does not close it.

### CODE REVIEW

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/HPackSpecExamples.scala [hunk 1]
```diff
@@ -4,7 +4,8 @@
 
 package akka.http.impl.engine.http2
 
-import akka.http.scaladsl.model.{ ContentTypes, HttpEntity, HttpProtocols, HttpResponse, headers }
+import akka.http.scaladsl.client.RequestBuilding.Get
```
Comment (johanandren) [inline akka-http2-support/src/test/scala/akka/http/impl/engine/http2/HPackSpecExamples.scala:7]: Intended add, no other changes in file?

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala [hunk 2]
```diff
@@ -192,6 +192,170 @@ class Http2ClientSpec extends AkkaSpecWithMaterializer("""
...
       }
 
+      "accept response with one HEADERS and one CONTINUATION frame" in new TestSetup with NetProbes {
+        user.emitRequest(0x1, Get("https://www.example.com/"))
+        network.expect[HeadersFrame]()
+
+        val headerBlock = HPackSpecExamples.C61FirstResponseWithHuffman
+        val fragment1 = headerBlock.take(8) // must be grouped by octets
+        val fragment2 = headerBlock.drop(8)
+        network.sendHEADERS(0x1, endStream = true, endHeaders = false, fragment1)
```
Comment (johanandren) [inline akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala:202]: I couldn't figure out why end-stream should not lead to an error when you send CONTINUATION from the spec, did I miss something?

**Review by johanandren — COMMENTED** [review]
Mostly LGTM, one question though.

File: akka-http-core/src/test/scala/akka/http/impl/engine/ws/ByteStringSinkProbe.scala [hunk 3]
```diff
@@ -108,4 +108,11 @@ private[http] object ByteStringSinkProbe {
 
       def within[T](max: FiniteDuration)(f: => T): T = probe.within(max)(f)
     }
+
+  def apply(source: Source[ByteString, _])(implicit system: ClassicActorSystemProvider): ByteStringSinkProbe = {
```
Comment (jrudolph) [inline akka-http-core/src/test/scala/akka/http/impl/engine/ws/ByteStringSinkProbe.scala:112]: Nice helper.

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala (hunk 2 above)
Comment (jrudolph) [reply akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala:202]: CONTINUATION does not define an END_STREAM flag so it needs to go to corresponding HEADERS flag. A few things seem to be botched around the CONTINUATION spec bits so do not lose too much sleep about that...

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala [hunk 4]
```diff
@@ -1318,66 +1318,11 @@ class Http2ServerSpec extends AkkaSpecWithMaterializer("""
     val network = new NetworkSide(fromNet, toNet, framesOut) with Http2FrameHpackSupport
   }
 
-  class NetworkSide(val fromNet: Probe[ByteString], val toNet: ByteStringSinkProbe, val framesOut: Http2FrameProbe) extends Http2FrameProbeDelegator with Http2FrameSending {
+  class NetworkSide(val fromNet: Probe[ByteString], val toNet: ByteStringSinkProbe, val framesOut: Http2FrameProbe) extends WindowTracking {
```
Comment (jrudolph) [inline akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala:1321]: Great, thanks!

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/WindowTracking.scala [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    } catch {
+      case e: AssertionError if e.getMessage contains "but only got [0] bytes" =>
+      // timeout, that's expected
+      case e: AssertionError if (e.getMessage contains "block took") && (e.getMessage contains "exceeding") =>
+        // pause like GC, poll again just to be sure
+        pollForWindowUpdates(duration)
+    }
+
+  // keep counters that are updated on outgoing network.sendDATA and incoming WINDOW_UPDATE frames
+  private var toServerWindows: Map[Int, Int] = Map.empty.withDefaultValue(Http2Protocol.InitialWindowSize)
```
Comment (jrudolph) [inline akka-http2-support/src/test/scala/akka/http/impl/engine/http2/WindowTracking.scala:50]: These names are now not valid anymore. Maybe rename all those with "server" in it to `toPeerWindow...`

**Review by jrudolph — APPROVED** [review]
LGTM, a few cosmetic things.

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala (hunk 2 above)
Comment (johanandren) [reply akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala:202]: Ah, thanks for the clarification.

**Review by johanandren — APPROVED** [review]

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/HPackSpecExamples.scala (hunk 1 above)
Comment (raboof) [reply akka-http2-support/src/test/scala/akka/http/impl/engine/http2/HPackSpecExamples.scala:7]: oops, reverting

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/WindowTracking.scala (hunk 5 above)
Comment (raboof) [reply akka-http2-support/src/test/scala/akka/http/impl/engine/http2/WindowTracking.scala:50]: Naming is hard! :D

This represents the window we're publishing for incoming data, so perhaps `windowForIncomingData`.

**Pull request merged**